flink-runtime/src/test/java/org/apache/flink/runtime/rpc/akka/AkkaRpcActorTest.java

```diff
@@ -358,6 +362,43 @@ public void testOnStopFutureCompletionDirectlyTerminatesAkkaRpcActor() throws Ex
 		}
 	}
 
+	/**
+	 * Tests that the {@link RpcEndpoint#onStart()} method is called when the {@link RpcEndpoint}
+	 * is started.
+	 */
+	@Test
+	public void testOnStartIsCalledWhenRpcEndpointStarts() throws Exception {
+		final CountDownLatch countDownLatch = new CountDownLatch(1);
+		final OnStartEndpoint onStartEndpoint = new OnStartEndpoint(akkaRpcService, countDownLatch, null);
```

Comment:
A nicer API might be:

```
final OnStartEndpoint onStartEndpoint = new OnStartEndpoint(akkaRpcService, countDownLatch);
try {
	onStartEndpoint.start();
	assertTrue(onStartEndpoint.isOnStartCalledWithin(timeout));
}
```

Reply:
I'll take a look to see whether I can simplify it that way.